kubernetes-membership-scheme/src/main/java/org/wso2/carbon/membership/scheme/kubernetes/resolver/DNSBasedPodIpResolver.java

```
@@ -0,0 +1,139 @@
+import org.wso2.carbon.utils.xml.StringUtils;
+import org.xbill.DNS.*;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.*;
+
+/**
+ * Class responsible for resolving pod ips based on DNS lookups
+ */
+public class DNSBasedPodIpResolver extends AddressResolver {
+
+    private static final Log log = LogFactory.getLog(DNSBasedPodIpResolver.class);
+    private int dnsLookupTimeout;
+
+    public DNSBasedPodIpResolver(Map<String, Parameter> parameters) throws KubernetesMembershipSchemeException {
+        super(parameters);
+        initialize();
+    }
+
+    private void initialize () throws KubernetesMembershipSchemeException {
+        String dnsLookupTimeoutValue = System.getenv(Constants.DNS_LOOKUP_TIMEOUT);
+        if (StringUtils.isEmpty(dnsLookupTimeoutValue)) {
+            dnsLookupTimeoutValue = getParameterValue(Constants.DNS_LOOKUP_TIMEOUT, "10");
+        }
+
+        dnsLookupTimeout = Integer.parseInt(dnsLookupTimeoutValue);
+    }
+
+    @Override
+    public Set<String> resolveAddresses() throws KubernetesMembershipSchemeException {
+
+        final Set<String> containerIPs = new HashSet<>();
+
+        for (String kubernetesService : getKubernetesServicesArray()) {
+            // lookup name format:
+            // <service-name>.<namespace>.svc.cluster.local
+            String dnsLookupName = new StringBuilder(kubernetesService.trim()).append(".").
+                    append(getKubernetesNamespace()).append(".").append("svc.cluster.local").toString();
+            log.info("Going to perform a DNS lookup for: " + dnsLookupName);
+
+            Lookup lookup = buildLookup(dnsLookupName);
+            Record[] records = lookup.run();
+
+            if (lookup.getResult() != Lookup.SUCCESSFUL) {
+                log.warn("DNS lookup for name '" + dnsLookupName + "' failed");
+                return Collections.emptySet();
```

Do we handle this condition in the caller? Wouldn't it better to throw an exception?